packages/react-core/src/components/Popover/Popover.tsx

```diff
@@ -89,251 +88,231 @@ export interface PopoverProps {
   position?: 'auto' | 'top' | 'bottom' | 'left' | 'right';
   /**
    * Callback function that is only invoked when isVisible is also controlled. Called when the popover Close button is
-   * clicked or the ESC key is used
+   * clicked, Enter key was used on it, or the ESC key is used
    */
-  shouldClose?: (tip: TippyInstance) => void;
+  shouldClose?: (hideFnc?: any) => void;
+  /**
+   * Callback function that is only invoked when isVisible is also controlled. Called when the Enter key is
+   * used on the focused trigger
+   */
+  shouldOpen?: (showFnc?: any) => void;
```

Comment:
Can we call these `onShouldClose` and `onShouldOpen`?

Reply:
the previous API already had `shouldClose` and as i wanted to break as few things as possible i kept it that way